### ISSUE TITLE
ReadError: The server aborted the pending request

### ISSUE THREAD

**Issue by Stono** [sindresorhus/got#1343]
#### Describe the bug

 - Node.js version: 14
 - OS & version: 

#### Actual behavior
I am seeing unhandled promises being thrown when a request times out, when using got >= `11.2.0`.  This is absolutely fine on `11.1.4`

```
Unhandled Rejection!
ReadError: The server aborted the pending request
    at PassThrough.<anonymous> (/tmp/app/node_modules/got/dist/source/core/index.js:684:31)
    at Object.onceWrapper (events.js:420:28)
    at PassThrough.emit (events.js:314:20)
    at IncomingMessage.<anonymous> (/tmp/app/node_modules/got/node_modules/mimic-response/index.js:59:12)
    at Object.onceWrapper (events.js:420:28)
    at IncomingMessage.emit (events.js:314:20)
    at IncomingMessage.origin.emit (/tmp/app/node_modules/@szmarczak/http-timer/dist/source/index.js:39:20)
    at TLSSocket.socketCloseListener (_http_client.js:426:11)
    at TLSSocket.emit (events.js:326:22)
    at net.js:673:12 {
  code: undefined,
  timings: {
    start: 1593863366791,
    socket: 1593863366791,
    lookup: 1593863366792,
    connect: 1593863366792,
    secureConnect: 1593863366793,
    upload: 1593863366793,
    response: 1593863366828,
    end: undefined,
    error: undefined,
    abort: undefined,
    phases: {
      wait: 0,
      dns: 1,
      tcp: 0,
      tls: 1,
      request: 0,
      firstByte: 35,
      download: undefined,
      total: undefined
    }
  }
}
```

It seems to be related to https://github.com/sindresorhus/got/issues/1308 and https://github.com/sindresorhus/got/issues/1311 - if you feel this is a duplicate, please feel free to close it.

#### Expected behavior
Timeouts to not be thrown as unhandled promises

**Comment by szmarczak** [comment]
Duplicate of #1308

**Comment by szmarczak** [comment]
Also released `got@11.4.0` :)

**Comment by Stono** [comment]
@szmarczak just tested the fix and it works :) thanks
